mdn/src/main/java/org/apache/james/mdn/MDN.java

```diff
@@ -206,9 +208,13 @@ public MimeMultipart asMultipart() throws MessagingException {
         multipart.setReportType(DISPOSITION_NOTIFICATION_REPORT_TYPE);
         multipart.addBodyPart(computeHumanReadablePart());
         multipart.addBodyPart(computeReportPart());
-        if (message.isPresent()) {
-            multipart.addBodyPart(computeOriginalMessagePart());
-        }
+        message.ifPresent(message1 -> {
+            try {
+                multipart.addBodyPart(computeOriginalMessagePart(message1));
+            } catch (MessagingException e) {
+                e.printStackTrace();
```

Comment:
Debug spotted.

We would write it

```
message.ifPresent(Throwing.consumer(originalMessage -> multipart.addBodyPart(computeOriginalMessagePart(originalMessage)).sneakyThrow());
```

To have the exception unchecked in the lambda...